superproxy/superproxy.go

```diff
@@ -48,6 +51,9 @@ type SuperProxy struct {
 
 	//usage
 	Usage *usage.ProxyUsage
+
+	//concurrency chan
+	concurrencyChan chan struct{}
```

Comment:
why struct? any better types?

Reply:
reference: https://dave.cheney.net/2014/03/25/the-empty-struct